src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -136,9 +146,20 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     configuration.setBoolean("fs.gs.impl.disable.cache", true);
     configuration.setBoolean("fs.gs.metadata.cache.enable", false);
 
+    if (config.getServiceAccountFilePath() != null) {
+      configuration.set(BigQueryConstants.CONFIG_SERVICE_ACCOUNT_FILE_PATH, config.getServiceAccountFilePath());
+    }
+    if (config.getPartitionFrom() != null) {
+      configuration.set(BigQueryConstants.CONFIG_PARTITION_FROM_DATE, config.getPartitionFrom());
+    }
+    if (config.getPartitionTo() != null) {
+      configuration.set(BigQueryConstants.CONFIG_PARTITION_TO_DATE, config.getPartitionTo());
+    }
+    configuration.set(BigQueryConstants.CONFIG_TEMPORARY_TABLE_NAME, uuid.toString().replaceAll("-", "_"));
```

Comment:
the temporary table name should be an implementation detail that only the InputFormat should know about. Allowing the plugin to set it would actually open up the possibility of the plugin using a non-unique name, resulting in failures.

Reply:
Done.